### PR TITLE
hotfix (app): Evm payload function arg parser update

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Fixes https://app.clickup.com/t/9002019994/PRF-7131

The functions with 0 arguments result into ETH_BAD_ARGUMENTS which is incorrect. Update the parser to return OK if all of the payload is parsed irrespective of the number of arguments.

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-7131 Complaint #905](https://app.clickup.com/t/9002019994/PRF-7131)

File: tests/unit_test_lists.c
Comment (ParnikaGupta9) [inline tests/unit_test_lists.c:1]: manual test is removed from these tests?

Comment (ujjwal-cyph) [reply tests/unit_test_lists.c:1]: no only moved around.

https://github.com/Cypherock/x1_wallet_firmware/pull/488/files#diff-847afcfe82be7431c58f82550f6595a6e15615b48c8ecc87eb91f5cb6a30a874R233

File: apps/evm_family/evm_contracts.c [hunk 1]
```diff
@@ -281,7 +281,13 @@ uint8_t ETH_ExtractArguments(const uint8_t *pAbiPayload,
...
   uint8_t currArgument;
 
-  for (currArgument = 0; currArgument < numArgsInFunction; currArgument++) {
+  for (currArgument = 0; currArgument <= numArgsInFunction; currArgument++) {
+    // only return OK when completely parsed whole payload
+    if ((pAbiPayload + sizeOfPayload) == pCurrHeadPtr) {
+      returnCode = ETH_UTXN_ABI_DECODE_OK;
+      break;
+    }
+
```
Comment (ParnikaGupta9) [inline apps/evm_family/evm_contracts.c:290]: The if condition can be outside the loop when no arguments are there.

File: apps/evm_family/evm_contracts.c [hunk 2]
```diff
@@ -281,7 +281,13 @@ uint8_t ETH_ExtractArguments(const uint8_t *pAbiPayload,
   const uint8_t *pPayloadBasePtr = pCurrHeadPtr;
   uint8_t currArgument;
 
-  for (currArgument = 0; currArgument < numArgsInFunction; currArgument++) {
+  for (currArgument = 0; currArgument <= numArgsInFunction; currArgument++) {
+    // only return OK when completely parsed whole payload
```
Comment (ParnikaGupta9) [inline apps/evm_family/evm_contracts.c:285]: The comment can mention the case where no arguments and/or why it is valid.

File: apps/evm_family/evm_contracts.c (hunk 1 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_contracts.c:288]: changed

File: apps/evm_family/evm_contracts.c (hunk 2 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_contracts.c]: updated

**Review by ParnikaGupta9 — APPROVED** [review]

**Review by ParnikaGupta9 — APPROVED** [review]

**Pull request merged**